### PR TITLE
Installing linter in the project

### PR DESCRIPTION
**Pull request by NikhilM98**
@llaske @tarunsinghal92 I have written the configuration for eslint. I've tried to minimize the errors.
Currently, it's detecting 278 errors. Out of which 9 can be automatically fixed. Should I go ahead and fix those errors manually?

Also, I need suggestions for linting `.ejs` files. I could not find any good plugin for linting them.

### CODE REVIEW

**Comment by tarunsinghal92** [comment]
@NikhilM98 

Could you categorize the errors? We can then discuss about the resolution.

**Comment by NikhilM98** [comment]
@tarunsinghal92 The majority of errors can be categorized and listed in the order of priority as:
- Undefined Variables.
- Variables already defined.
- Variable defined but never used.
- Unnecessary/Missing Semicolon.
- Mixed spaces and tabs.
- Indentation. Linebreaks.

**Comment by llaske** [comment]
Look like minor errors.

**Comment by NikhilM98** [comment]
@llaske the errors are minor but if we merge the linter configuration then it will detect 8250 such minor errors and project will appear red with errors.
If I fix all those errors at once. It will be difficult to review the changes and the git history won't look clean.
- Fix all errors at once. Difficult to review.
- Handle one type of errors in the project in one commit. Easier to review.
- Handle batches of files at once. Like fixing api-controller in one go, dashboard-controller in the next. It will also be easier to review and keep the git history clean.

**Comment by tarunsinghal92** [comment]
`Handle one type of errors in the project in one commit` Don't push more till the previous one got approved.

**Comment by llaske** [comment]
I'm not in favor to implement this feature now.
It require too much change in current source code without a real benefit.

**Comment by NikhilM98** [comment]
@llaske too many changes are the reason why I wanted to discuss the implementation with you.

I think we can install the linter with only basic configuration to lint errors related to the undefined, defined but not used and use of es6/es7 syntax. It can be achieved with very less changes in the code.

As the project will grow larger, someday or the other we've to integrate linter, codacy, travis, etc. in this project. Installing linter with basic configuration can be the first step towards that. Not having coding conventions is also a barrier towards attracting contributions.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
+dashboard/public/js/bootstrap-notify.js
+dashboard/public/js/bootstrap.min.js
+dashboard/public/js/Chart.min.js
+dashboard/public/js/jquery-3.1.0.min.js
+dashboard/public/js/jquery-sortable.js
+dashboard/public/js/jquery-ui-1-11-4.js
+dashboard/public/js/jquery.multi-select.js
+dashboard/public/js/jquery.searchable.js
```
Comment (tarunsinghal92) [inline .eslintignore]: `cant you just use dashboard/public/*`

File: package.json [hunk 2]
```diff
@@ -31,13 +31,15 @@
...
+		"eslint": "^5.15.0"
 	},
 	"scripts": {
 		"test": "mocha api/test/",
 		"docs": "apidoc -i api/controller -i dashboard/helper -o docs/www/",
 		"start": "node sugarizer.js",
 		"start:watch": "nodemon sugarizer.js",
-		"seed:users-csv": "node scripts/seed_users.js"
+		"seed:users-csv": "node scripts/seed_users.js",
+		"lint": "eslint .; exit 0"
```
Comment (tarunsinghal92) [inline package.json]: why do we need `exit 0`

Comment (NikhilM98) [reply package.json]: If ESlint detects errors in the file, it exits with error code 1. The output looks like the node script is crashing. I've used `exit 0` to suppress the error.

File: .eslintignore (hunk 1 above)
Comment (NikhilM98) [reply .eslintignore]: We need to lint files such as `main.js` which is stored in the same directory.

P.S. I guess we can further restructure the project to separate public resources with project js files

File: package.json (hunk 2 above)
Comment (tarunsinghal92) [reply package.json]: yeah you can format this file https://github.com/llaske/sugarizer-server/blob/master/dashboard/public/js/main.js and move code to respective controllers.

File: .eslintignore (hunk 1 above)
Comment (tarunsinghal92) [reply .eslintignore]: yeah you can format this file https://github.com/llaske/sugarizer-server/blob/master/dashboard/public/js/main.js and move code to respective controllers.

**Pull request merged**